lib/elasticbeats.py

```diff
@@ -44,10 +45,15 @@ def render_without_context(source, target):
 
 
 def principal_unit_cache():
-    principal_name = getenv('JUJU_REMOTE_UNIT')
     cache = kv()
+    principal_name = getenv('JUJU_PRINCIPAL_UNIT')
+    if principal_name == '':
+        principal_name = getenv('JUJU_REMOTE_UNIT')
```

Comment:
I know we had `REMOTE_UNIT` before this proposal, but I don't think that is ever the right answer when trying to find our principal.  I think we should drop this and alter the condition to be something like:

```
if principal_name:
    cache.set('principal_name', principal_name)
```